# DP: Fix PR target/70496 (ARM32), taken from the trunk

gcc/

2016-04-01  Ramana Radhakrishnan  <ramana.radhakrishnan@arm.com>

	PR target/70496
	* config/arm/arm.h (ASM_APP_OFF): Handle TARGET_ARM
	and TARGET_THUMB.

gcc/testsuite/

2016-04-01  Ramana Radhakrishnan  <ramana.radhakrishnan@arm.com>

	PR target/70496
	* gcc.target/arm/pr70496.c: New test.

 
--- a/src/gcc/testsuite/gcc.target/arm/pr70496.c
+++ b/src/gcc/testsuite/gcc.target/arm/pr70496.c
@@ -0,0 +1,12 @@
+/* { dg-do assemble } */
+/* { dg-options "-mthumb -O2" } */
+/* { dg-require-effective-target arm_thumb2_ok } */
+
+int i;
+void
+main (void)
+{
+  __asm__ volatile (".arm");
+  i = 0;
+  __asm__ volatile ("\n cbz r0, 2f\n2:");
+}
--- a/src/gcc/config/arm/arm.h
+++ b/src/gcc/config/arm/arm.h
@@ -2037,7 +2037,8 @@
 		    "\t.syntax divided\n")
 
 #undef  ASM_APP_OFF
-#define ASM_APP_OFF "\t.syntax unified\n"
+#define ASM_APP_OFF (TARGET_ARM ? "\t.arm\n\t.syntax unified\n" : \
+		     "\t.thumb\n\t.syntax unified\n")
 
 /* Output a push or a pop instruction (only used when profiling).
    We can't push STATIC_CHAIN_REGNUM (r12) directly with Thumb-1.  We know
